include/NZSL/Ast/Nodes.hpp

```diff
@@ -117,10 +117,18 @@ namespace nzsl::Ast
 
 	struct NZSL_API CallFunctionExpression : Expression
 	{
+		enum class ParameterSemantic : Nz::UInt8
+		{
+			In,
+			Out,
+			InOut,
+		};
+
 		NodeType GetType() const override;
 		void Visit(ExpressionVisitor& visitor) override;
 
 		std::vector<ExpressionPtr> parameters;
+		std::vector<ParameterSemantic> parametersSemantic;
```

Comment:
I think they should be part of the same vector, as each parameter will have its own semantic